src/admin.rs

```diff
@@ -353,39 +325,76 @@ impl PgqrsAdmin {
         Ok(results)
     }
 
-    /// Delete a queue and all its messages from the database.
-    /// This also deletes the corresponding archive table.
+    /// Delete a queue from the database.
+    ///
+    /// A queue can only be deleted if there are no references in messages or archive.
```

Comment:
A queue should also not be deleted if there are running or shutting down workers.